src/plugins/discover/public/application/main/hooks/use_adhoc_data_views.test.tsx

```diff
@@ -104,57 +108,31 @@ describe('useAdHocDataViews', () => {
       ...mockDataView,
       id: 'updated-mock-id',
     }));
-    const hook = renderHook((d: DataView) =>
-      useAdHocDataViews({
-        dataView: mockDataView,
-        savedSearch: savedSearchMock,
-        stateContainer: {
-          appStateContainer: { getState: jest.fn().mockReturnValue({}) },
-          replaceUrlAppState: jest.fn(),
-          kbnUrlStateStorage: {
-            kbnUrlControls: { flush: jest.fn() },
-          },
-        } as unknown as GetStateReturn,
-        setUrlTracking: jest.fn(),
-        dataViews: mockDiscoverServices.dataViews,
-        filterManager: mockDiscoverServices.filterManager,
-        toastNotifications: mockDiscoverServices.toastNotifications,
-      })
+    const stateContainer = getDiscoverStateMock({ isTimeBased: true });
+    const hook = renderHook(
+      () =>
+        useAdHocDataViews({
+          dataView: mockDataView,
+          savedSearch: savedSearchMock,
+          stateContainer: getDiscoverStateMock({ isTimeBased: true }),
+          setUrlTracking: jest.fn(),
+          dataViews: mockDiscoverServices.dataViews,
+          filterManager: mockDiscoverServices.filterManager,
+          toastNotifications: mockDiscoverServices.toastNotifications,
+        }),
+      {
+        wrapper: ({ children }: { children: React.ReactElement }) => (
+          <DiscoverMainProvider value={stateContainer}>{children}</DiscoverMainProvider>
+        ),
+      }
     );
 
     let updatedDataView: DataView;
     await act(async () => {
       updatedDataView = await hook.result.current.updateAdHocDataViewId(mockDataView);
     });
 
-    expect(dataViewsCreateMock).toHaveBeenCalledWith({
-      id: undefined,
-    });
+    expect(mockDiscoverServices.dataViews.clearInstanceCache).toHaveBeenCalledWith(mockDataView.id);
     expect(updatedDataView!.id).toEqual('updated-mock-id');
   });
-
-  it('should update the adHocList correctly for text based mode', async () => {
```

Comment:
this is list is no longer stored in this hook, moved to internal state, that's why I removed this test